src/bun.js/bindings/JSBufferList.cpp

```diff
@@ -86,9 +93,15 @@ JSC::JSValue JSBufferList::join(JSC::VM& vm, JSC::JSGlobalObject* lexicalGlobalO
     }
     const bool needSeq = seq->length() != 0;
     const auto end = m_deque.end();
+    size_t offset = m_read;
     JSRopeString::RopeBuilder<RecordOverflow> ropeBuilder(vm);
     for (auto iter = m_deque.begin();;) {
         auto str = iter->get().toString(lexicalGlobalObject);
+        if (offset > 0) {
+            const size_t len = str->length() - offset;
+            str = JSC::jsSubstring(lexicalGlobalObject, str, offset, len);
```

Comment:
this needs a RETURN_IF_EXCEPTION in the extremely rare case where you ran out of memory and so jsSubstring threw

Reply:
https://github.com/oven-sh/bun/pull/1735/commits/e56d73e1d58cdb981d76d7f522bb02b49e2a33c1